### PR TITLE
fix: implement temporary schema definition to support datetime scalars

### PR DESCRIPTION
**Pull request by morozj01**
# Ensure DateTime scalar inputs are converted to string - #53 

## Description
This is meant to be a temporary solution to enable datetime scalar types in Ceramic models while [this pull request](https://github.com/Urigo/graphql-scalars/pull/1641) is in review (as there has been no activity for some time.)

This change introduces a custom datetime scalar definition in a new file which is meant to be deleted after the solution is merged into the [graphql-scalars package](https://www.npmjs.com/package/graphql-scalars)

This schema definition is essentially a simplified copy of the logic defined in the [graphql-scalars package](https://github.com/Urigo/graphql-scalars/blob/master/src/scalars/iso-date/DateTime.ts).

It purposely kept minimal and therefore only ISO strings in UTC time are supported - as per the [documentation](https://composedb.js.org/docs/0.3.x/guides/creating-composites/scalars#datetime)

Passing in Unix timestamp or JS date object will throw an error.

## How Has This Been Tested?
Describe the tests that you ran to verify your changes. Provide instructions for reproduction.

Currently creating documents with a datetime scalar field will throw [this error](https://forum.ceramic.network/t/cbor-encode-error-unsupported-type-date-when-trying-to-create-composedb-document/360/3)

With this change, I was able to upload datetime scalar values without issue

IDs: 
- kjzl6kcym7w8y5f7jspauwxecxeuzrri0rigwwl5h7x8i9ebhmh7hi2thxmksx0
- kjzl6kcym7w8y60p5d46t282vvl4b8t4pkfk1364rlhbxtqhb71i2rtsckvqrvb
- kjzl6kcym7w8y5u9ktpponjvo0jh0pjl54tc9tx0pl3ayupzxaamdb7vpyduc0y

Model: kjzl6hvfrbw6c65z7bpyak6mgz9maadbucil8sey0lkuy9ebiinbjl0vd9ygal4

_**Let me know if/what additional testing is needed**_

## Definition of Done

Before submitting this PR, please make sure:

- [x] The work addresses the description and outcomes in the issue
- [ ] I have added relevant tests for new or updated functionality
- [x] My code follows conventions, is well commented, and easy to understand
- [x] My code builds and tests pass without any errors or warnings
- [x] I have tagged the relevant reviewers
- [ ] I have updated the READMEs of affected packages
- [ ] I have made corresponding changes to the documentation
- [x] The changes have been communicated to interested parties

### CODE REVIEW

File: packages/graphql-scalars/src/datetime.ts [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+  parseLiteral: (ast) => {
+    if (ast.kind !== Kind.STRING) {
+      throw new TypeError('DateTime cannot represent non string or Date type')
+    }
+    const { value } = ast
+    if (validateDateTime(value)) {
+      return value
+    }
+    throw new TypeError(`DateTime cannot represent an invalid date-time-string ${String(value)}.`)
+  },
```
Comment (PaulLeCam) [inline packages/graphql-scalars/src/datetime.ts:99]: Can you add the `extensions` object from https://github.com/Urigo/graphql-scalars/blob/master/src/scalars/iso-date/DateTime.ts#L70-L76 as well please to ensure the JSON schema is available?

File: packages/graphql-scalars/src/datetime.ts [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    /^(\d{4}-(0[1-9]|1[012])-(0[1-9]|[12][0-9]|3[01])T([01][0-9]|2[0-3]):([0-5][0-9]):([0-5][0-9]|60))(\.\d{1,})?(([Z])|([+|-]([01][0-9]|2[0-3]):[0-5][0-9]))$/
+
+  // Validate the structure of the date-string
+  if (!RFC_3339_REGEX.test(input)) {
+    return false
+  }
+
+  // Check if it is a correct date using the javascript Date parse() method.
+  const time = Date.parse(input)
+  if (time !== time) {
```
Comment (PaulLeCam) [inline packages/graphql-scalars/src/datetime.ts:54]: This seems incorrect, this condition will always be `false`

File: packages/graphql-scalars/src/datetime.ts [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+    'A date-time string at UTC, such as 2007-12-03T10:15:30Z, ' +
+    'compliant with the date-time format outlined in section 5.6 of ' +
+    'the RFC 3339 profile of the ISO 8601 standard for representation ' +
+    'of dates and times using the Gregorian calendar.',
+  serialize: (dateString) => {
+    if (typeof dateString === 'string' && validateDateTime(dateString)) {
+      if (dateString.indexOf('Z') !== -1) {
+        return dateString
+      } else
+        throw new TypeError(`DateTime string must be formatted to UTC time  ${String(dateString)}.`)
```
Comment (PaulLeCam) [inline packages/graphql-scalars/src/datetime.ts:78]: There are 2 spaces before `${String(dateString)}`, can you remove one please?

File: packages/graphql-scalars/src/datetime.ts [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+    } else
+      throw new TypeError(
+        `DateTime cannot represent an invalid date-time-string  ${String(dateString)}.`
```
Comment (PaulLeCam) [inline packages/graphql-scalars/src/datetime.ts:81]: There are 2 spaces before `${String(dateString)}`, can you remove one please?

**Review by PaulLeCam — CHANGES_REQUESTED** [review]
Thanks for your PR!
Looks good to me overall, just a few comments.

File: packages/graphql-scalars/src/datetime.ts (hunk 2 above)
Comment (morozj01) [reply packages/graphql-scalars/src/datetime.ts]: I think it is supposed to check for NaN but totally agree it should be more explicit

**Review by PaulLeCam — APPROVED** [review]
Thanks for the changes!

**Comment by PaulLeCam** [comment]
@morozj01 could you update the scalar description to use the one from graphql-scalars so it matches the existing test snapshots please?

**Comment by morozj01** [comment]
> @morozj01 could you update the scalar description to use the one from graphql-scalars so it matches the existing test snapshots please?

For sure - just updated

**Review by PaulLeCam — APPROVED** [review]
Thanks!

**Pull request merged**